include/configs/imx8mm_evk.h

```diff
@@ -12,8 +12,10 @@
 
 #define CONFIG_SPL_MAX_SIZE		(148 * 1024)
 #define CONFIG_SYS_MONITOR_LEN		SZ_512K
+#ifndef CONFIG_SECONDARY_BOOT_RUNTIME_DETECTION
 #define CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_USE_SECTOR
 #define CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR	(0x300 + CONFIG_SECONDARY_BOOT_SECTOR_OFFSET)
```

Comment:
Not sure why we are setting this to 0x300 + CONFIG_SECONDARY_BOOT_SECTOR_OFFSET when this is #ifndef CONFIG_SECONDARY_BOOT_RUNTIME_DETECTION

Reply:
Ah.. I see.  Original intent of `CONFIG_SECONDARY_BOOT_SECTOR_OFFSET` was to be a compile time setting for u-boot.  We want to avoid this case when we are using RUNTIME detection.

Ignore this.

Reply:
Or maybe add a comment above the #ifndef to explain the intention.

Reply:
Actually, I would like to see this here (leaving all settings as defined but with correct values):
```
#define CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_USE_SECTOR
/*
 * CONFIG_SECONDARY_BOOT_SECTOR_OFFSET was intended to be used as a compile time setting.
 * If CONFIG_SECONDARY_BOOT_RUNTIME_DETECTION is defined, then only use the base offset of 0x300 here.
 * CONFIG_SECONDARY_BOOT_SECTOR_OFFSET will be added dynamically if needed.
 */
#ifdef CONFIG_SECONDARY_BOOT_RUNTIME_DETECTION
#define CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR	(0x300)
#else
#define CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR	(0x300 + CONFIG_SECONDARY_BOOT_SECTOR_OFFSET)
#endif /* CONFIG_SECONDARY_BOOT_RUNTIME_DETECTION */
```